### ISSUE TITLE
Can't use other lightcurve class methods after using bin() function

### ISSUE THREAD

**Issue by greenstar54** [KeplerGO/lightkurve#705]
#### Problem description

I am working with the PDCSAP_FLUX for TESS light curves. When I attempt to first fill the gaps in the light curve and then bin it using Lightkurve functions, I receive a type error like so:

TypeError: No loop matching the specified signature and casting
was found for ufunc bitwise_or

#### Example

```python
import lightkurve
lcf = lightkurve.search.open(filename)
lc = lcf.PDCSAP_FLUX

lc_filled = lc.fill_gaps()
lc_binned = lc_filled.bin(binsize=15)

```

#### Expected behavior

If I understand the docs correctly, both lc.fill_gaps() and lc.bin() should return lightcurve objects and thus the other methods in the class should work on it. However, attempting this returns a type error. I tried .bin(), .normalize(), and .remove_nans() which all gave me the same error.

TypeError: No loop matching the specified signature and casting
was found for ufunc bitwise_or

**Comment by barentsen** [comment]
Oops, this looks like a serious bug!

Slightly simplified snippet to reproduce the problem:
```python
import lightkurve as lk
import numpy as np
lc = lk.TessLightCurve(time=np.arange(50), flux=np.ones(50), quality=np.zeros(50))
lc.bin(binsize=15)
```

This yields the following exception:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-ecebd391bc07> in <module>
      2 import numpy as np
      3 lc = lk.TessLightCurve(time=np.arange(50), flux=np.ones(50), quality=np.zeros(50))
----> 4 lc.bin(binsize=15)

~/dev/lightkurve/lightkurve/lightcurve.py in bin(self, binsize, bins, method)
   1000                 warnings.simplefilter("ignore", RuntimeWarning)
   1001                 binned_stat = binned_statistic(bin_by_array, values_to_bin,
-> 1002                     statistic=bin_function, bins=bin_edges).statistic
   1003                 setattr(binned_lc, attr, binned_stat)
   1004 

~/bin/anaconda/lib/python3.7/site-packages/scipy/stats/_binned_statistic.py in binned_statistic(x, values, statistic, bins, range)
    179 
    180     medians, edges, binnumbers = binned_statistic_dd(
--> 181         [x], values, statistic, bins, range)
    182 
    183     return BinnedStatisticResult(medians, edges[0], binnumbers)

~/bin/anaconda/lib/python3.7/site-packages/scipy/stats/_binned_statistic.py in binned_statistic_dd(sample, values, statistic, bins, range, expand_binnumbers)
    604         for i in np.unique(binnumbers):
    605             for vv in xrange(Vdim):
--> 606                 result[vv, i] = statistic(values[vv, binnumbers == i])
    607 
    608     # Shape into a proper matrix

~/dev/lightkurve/lightkurve/lightcurve.py in <lambda>(x)
    952         method_func = np.__dict__['nan' + method]
    953         quality_func = lambda x: np.bitwise_or.reduce(x) \
--> 954                                  if np.all(np.isfinite(x)) else np.nan
    955         centroid_func = lambda x: method_func(x) \
    956                                   if np.any(np.isfinite(x)) else np.nan

TypeError: No loop matching the specified signature and casting
was found for ufunc bitwise_or
```

**Comment by barentsen** [comment]
This bug was likely introduced in Lightkurve v1.5.0 via #629.

**Comment by barentsen** [comment]
Findings: the `lc.bin()` method fails when `lc.quality` has a non-integer dtype, which is unusual but allowed.

There is a related bug in `lc.fill_gaps()` which causes the dtype of `lc.quality` to be turned from integer into floating point.  This explains why the bug in `bin()` only appears after calling `fill_gaps()`.

This will be easy to fix.

**Comment by barentsen** [comment]
This issue should be fixed now.  A fix and regression test were added in #707 

Thanks for reporting this bug @greenstar54! A valuable contribution!! :+1::+1:
